src/pods/BasicPodManager.ts

```diff
@@ -0,0 +1,62 @@
+import type { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import type { ResourceStore } from '../storage/ResourceStore';
+import { ConflictHttpError } from '../util/errors/ConflictHttpError';
+import { NotFoundHttpError } from '../util/errors/NotFoundHttpError';
+import { UnsupportedHttpError } from '../util/errors/UnsupportedHttpError';
+import { ensureTrailingSlash } from '../util/PathUtil';
+import type { DataGenerator } from './DataGenerator';
+import type { PodManager } from './PodManager';
+import type { User } from './User';
+
+/**
+ * Pod manager that uses a {@link DataGenerator} to create the default resources for a new pod.
+ */
+export class BasicPodManager implements PodManager {
+  private readonly base: string;
+  private readonly store: ResourceStore;
+  private readonly generator: DataGenerator;
+
+  public constructor(base: string, store: ResourceStore, generator: DataGenerator) {
+    this.base = base;
+    this.store = store;
+    this.generator = generator;
+  }
+
+  /**
+   * Creates a new pod using all resources created by the data generator.
+   * Pod identifiers are created by appending the username to the base URI.
+   * Will throw an error if the given identifier already has a resource.
+   */
+  public async createPod(user: User): Promise<ResourceIdentifier> {
+    const identifier = this.generateIdentifier(user.username);
+    try {
```

Comment:
Not urgent for this PR, but I guess some store implementations would be able to optimize this existence operation, so perhaps we'd want to add this to the store interface in the future.

Reply:
This is a discussion that has come up several times already, and the amount of classes we have that need a simple existence (or in some cases, only metadata) check does keep increasing. So there is something to be said for that.

Reply:
Agreed: https://github.com/solid/community-server/issues/373